tensorboard/components/tf_trace_viewer/tf-trace-viewer.html

```diff
@@ -97,44 +113,229 @@
           var components = parts[i].split('=');
           if (components[0] == "trace_data_url") {
             this.traceDataUrl = decodeURIComponent(components[1]);
-            break;
+          } else if (components[0] == "is_streaming") {
+            this._isStreaming = components[1] === 'true';
           }
         }
       }
 
+      if (!this.traceDataUrl) {
+        this._displayOverlay("Trace data URL is not provided.", "Trace Viewer");
+        return null;
+      }
       this._throbber.className = "active";
+
       this._loadTrace();
     },
+
+    // Something has changed, so consider reloading the data:
+    //   - if we have zoomed in enough to need more details.
+    //   - if we have scrolled too close to missing data regions
+    // We ensure there's only ever one request in flight.
     _loadTrace : function() {
-      if (!this.traceDataUrl) {
-        this._displayOverlay("Trace data URL is not provided.", "Trace Viewer");
-        return null;
+      if (!this._isStreaming) {
+        // Send HTTP request to get the trace data.
+        var req = new XMLHttpRequest();
+        var is_binary = / [.] gz$ /.test(this.traceDataUrl) ||
+                        / [.] zip$ /.test(this.traceDataUrl);
+        req.overrideMimeType('text/plain; charset=x-user-defined');
+        req.open('GET', this.traceDataUrl, true);
+        if (is_binary) {
+          req.responseType = 'arraybuffer';
+        }
+
+	req.onreadystatechange = event => {
+          if (req.readyState !== 4) {
+            return;
+          }
+	  window.setTimeout(() => {
+            if (req.status === 200) {
+              this._throbber.className = "inactive";
+              this.set("_traceData", is_binary ? req.response : req.responseText);
```

Comment:
Why is binary and textual responses handled by the same business logic? Does this XHR really fetch both the HTML page in addition to streaming binary data? What sort of streaming binary data is it fetching?

Reply:
I guess is_binary never set to true in the old code;  and I feel uncomfortable to change it right now in this CL.
my intention is to maintain backward compatibilities.

Reply:
done. removed.